src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -207,9 +292,65 @@ export const ChannelView: React.FC = () => {
 
 const usePagination = (currentTab: number) => {
   const [currentPage, setCurrentPage] = useState(0)
+  const [currentSearchPage, setCurrentSearchPage] = useState(0)
   // reset the pagination when changing tabs
   useEffect(() => {
     setCurrentPage(0)
   }, [currentTab])
-  return { currentPage, setCurrentPage }
+  useEffect(() => {
+    setCurrentSearchPage(0)
+  }, [currentSearchPage])
+  return { currentPage, setCurrentPage, currentSearchPage, setCurrentSearchPage }
+}
+
+type useSearchVideosParams = {
+  id: string
+}
+const useSearchVideos = ({ id }: useSearchVideosParams) => {
```

Comment:
What's the exact purpose of this hook? I understand that it's supposed to make the component look cleaner and extract some logic out of it. It seems though that it's not really defined what goes into it and what doesn't. For example, search state and refs are defined here but the keypress is defined in the component. I suggest you figure out what you want from this hook and clean this up